File: Tests/EndToEnd/Classes/declarations.java
-
class Point {
    public int x;
    public int y;
    protected int square;
    private string name;
    
    Point(int x, int y) {
        this.x = x;
        this.y = y;
        this.square = x * y;
        this.name = "";
    }
}

class Measurements<T> {
    Measurements(T[] items) {
        // ...
    }
}

class Shape extends Measurements<Point> {
    Shape(Point[] points) {
        // ...
    }
}
-

File: Tests/EndToEnd/Classes/declarations.rb
-
class Point
    
    def initialize(x, y)
        self.x = x
        self.y = y
        self.square = x * y
        self.name = ""
    end
end

class Measurements
    def initialize(items)
        # ...
    end
end

class Shape < Measurements
    def initialize(points)
        # ...
    end
end
-
